### ISSUE TITLE
WiFi registration helper & fake heartbeat server

### ISSUE THREAD

**Issue by markv9401** [RobHofmann/HomeAssistant-GreeClimateComponent#131]
**Is your feature request related to a problem? Please describe.**
For me, a 2022 series Syen AC owner, it is a problem. Without a heartbeat server the HVAC unit will eventually 'reset' itself and lose the app / HA.

**Describe the solution you'd like**
Based on existing solutions, repositories and some WireSharking I've hacked together my own solution that does not require letting the HVAC unit connect to the actual WAN internet at all, ever.

**Additional context**
Do you think it'd be feasible to try and merge a fake heartbeat server into this integration? I don't believe it should be too difficult and it could be useful not having to run it separately. Please take a look at it (  https://github.com/markv9401/Fake-Gree-server  ) and if you think it could be useful, I'd happily try to merge it into the server in a PR. Thanks!

**Comment by maciejtarmas** [comment]
While that would be in fact useful, your solution doesn't work for my 3 Cooper & Hunter units (basically another Gree brand). Emtek's does.

I don't know what the code actually does, but I noticed that you're exposing port 1812, when most known Gree variants connect to port 5000. Changing the port in your implementation doesn't do anything in my case though.

This might need more investigation.

As to WiFi registration, I wouldn't bother. You can spin up the fake heartbeat server, block all outgoing WAN traffic from the unit, reset it and try to register with the app again. The app will tell you that it failed (because it can't talk to China at this point, obviously), but the AC unit will keep the WiFi credentials that you entered.

**Comment by markv9401** [comment]
Sure thing, I don't mind, just thought it could be useful. 
Port 1812 explanation: During the initial registration process _(when you reset WiFi and first log in to Gree servers)_ it uses an apparently burnt-in domain and port, 1812. The response to this requests contains the servers used in further requests, including the port. That usually (exclusively?) ends up being port 5000. But it works just fine, the HVAC unit will work with whatever port it receives in the response. So by trying to include registration and changing the port to 1812, you essentially need a single port not 2 of them.

**Comment by rapi3** [comment]
Thank you for your work !
I  run the python fake server script to an existing Gree AC setup already registered to Gree server.
From firewall I redirect all traffic to wan ports 1812 and 5000 from LAN Gree ip to my server running your script and block everything else to WAN from Gree AC and no problem.
All it is good.

**Comment by Obscurax** [comment]
I've tried this but it does not work with, Vaillant WiFi Adapter VAI8/5-i.
Note that this device is supported since encryption v2 is supported.

I've set up https://github.com/markv9401/Fake-Gree-server
Also tried https://github.com/emtek-at/GreeAC-DummyServer

A few minutes after traffic is blocked from the AC to outside the connection between the AC and HA is lost.
It'd be nice to find a way to block all home calling.

**Comment by jodewee** [comment]
> e that this device is supported since encryption v2 is suppo

I have about the same (bulex vivair, but basically the same)
no need for fake gree server.
just add the v2 encryption method in the config.yaml

**Comment by Obscurax** [comment]
> > e that this device is supported since encryption v2 is suppo
> 
> I have about the same (bulex vivair, but basically the same) no need for fake gree server. just add the v2 encryption method in the config.yaml

Sorry but this is not about the component not working. Which it does. 
It's about blocking home calling from the AC to the Gree servers.

**Comment by markv9401** [comment]
Hey,
It's been a minute ago since I had put this together but it's still working fine with my Syen (Gree) AC just fine. I haven't had to re-register a single time despite power outages etc. Basically what I'm saying is that it was and still is working with Gree / Syen, or with the same wifi controllers as mine. I can try taking a look into yours and help you debug / help you find out what we need to change about the code / protocol but I'll need a packet capture dump to see what could be missing or what's wrong. Can you record a session?

**Comment by RobHofmann** [comment]
I need to look into this. I realize that this might be the reason that my Gree HVAC's seem to disconnect from WiFi from time to time.

Im not sure when i'll have for this, but it's interesting nonetheless.

**Comment by RobHofmann** [comment]
The behaviour im seeing here is that randomly the ac will lose connection to the wifi. This happend every X days or weeks randomly. Then i need to onboard the device from scratch again.

**Comment by RobHofmann** [comment]
Yeah it started happening last year somewhere. My HVAC's have always been (like 5+ years) in an internet disconnected environment. Always worked just fine and all of a sudden they started to lose connection to the wifi.

Im positive it had to do with a WiFi firmware upgrade or setting somehow since the HVAC's havent changed.

**Comment by RobHofmann** [comment]
Right. Im running the latest version on mine, but thats a quite old version already.

**Comment by RobHofmann** [comment]
@simonepsp feel free to make a PR on this repo where we explain that you can use this add-on for this purpose :).
